pkg/controller/tuned/tuned_controller.go

```diff
@@ -122,6 +179,20 @@ func (r *ReconcileTuned) syncServiceAccount(tuned *tunedv1.Tuned) error {
 	} else {
 		// Do not update service account as it leads to accumulating secrets; see BZ1714484
 		glog.V(2).Infof("Tuned ServiceAccount already exists, skipping update")
+
+		isUpgrade, err := r.isOperandUpgrade(tuned)
+		if err != nil {
+			glog.Errorf("%v", err)
+			return nil
+		}
+		if isUpgrade {
+			// This is really only necessary to do operator-assisted removal of accumulated secrets prior to commit 014900fe
+			glog.V(2).Infof("This is an operand upgrade, removing all tuned secrets.")
+			err := r.removeTunedSecrets(tuned)
+			if err != nil {
+				glog.Errorf("Failed to remove tuned secrets: %v", err)
```

Comment:
Do we want to `return err` after this as well?

Reply:
Don't think we do.  The reason as above.